lib/util/version.js

```diff
@@ -12,7 +13,23 @@ function getReactVersionFromContext(context) {
   let confVer = '999.999.999';
   // .eslintrc shared settings (http://eslint.org/docs/user-guide/configuring#adding-shared-settings)
   if (context.settings.react && context.settings.react.version) {
-    confVer = context.settings.react.version;
+    let settingsVer = context.settings.react.version;
+    if (settingsVer === 'detect') {
+      try {
+        const reactPath = resolve.sync('react', {basedir: process.cwd()});
+        const react = require(reactPath);
+        settingsVer = react.version;
+      } catch (e) {
+        if (e.code === 'MODULE_NOT_FOUND') {
+          log('Warning: React version was set to "detect" in eslint-plugin-react settings, ' +
+            'but the "react" package is not installed. Assuming latest React version for linting.');
+          settingsVer = '999.999.999';
```

Comment:
this should probably set `warnedForMissingVersion = true` as well?

Reply:
it can, but even without that execution would never go into the branch where we warn because we do have a value in `context.settings.react.version`.

Reply:
ah, fair enough